ghcide/src/Development/IDE/Core/Shake.hs

```diff
@@ -568,15 +569,13 @@ deleteValue ShakeExtras{dirtyKeys, state} key file = do
     modifyTVar' dirtyKeys $ insertKeySet (toKey key file)
 
 recordDirtyKeys
-  :: Shake.ShakeValue k
-  => ShakeExtras
-  -> k
-  -> [NormalizedFilePath]
+  :: ShakeExtras
+  -> [Key]
   -> STM (IO ())
-recordDirtyKeys ShakeExtras{dirtyKeys} key file = do
-    modifyTVar' dirtyKeys $ \x -> foldl' (flip insertKeySet) x (toKey key <$> file)
+recordDirtyKeys ShakeExtras{dirtyKeys} keys = do
```

Comment:
I don't think this function serves much of a purpose anymore, it can be deleted. The trace isn't very important.